### PR TITLE
Bump ReactiveSwift to 3.0

### LINKED ISSUE #1468: Update ReactiveSwift dependency
there is 3.0.0 version released, how about updating Moya dependency to 3.0.0 as well? (this is the only thing which stops me from updating my project to new pod versions)

### PR DESCRIPTION
**Pull request by larryonoff**
Fixes #1468
This PR contains ReactiveSwift update to 3.0

### CODE REVIEW

**Comment by SD10** [comment]
We may have to update the `Package.swift` as well. I'm also unsure if this should go into `master` or `development`. If `master`, we could do a minor patch release including the bugfixes we've merged.

**Comment by larryonoff** [comment]
@sd10 sure! Better in this PR or separate ?

**Comment by larryonoff** [comment]
Some linter errors fixed in a separate PR #1473

**Comment by larryonoff** [comment]
Closed separate PR with linter fixes and made the change in the current only

**Comment by SD10** [comment]
I've changed this from `master` to `development` because this change will go into Moya 11.0. The excess commits look scary but it's only because we need to merge the commits currently on `master` into `development`. Then they should all disappear.

**Comment by sunshinejr** [comment]
Hey @larryonoff, could you rebase your PR, please? I think we will be looking at new major version in the near future 😉

**Comment by larryonoff** [comment]
@sunshinejr Just rebased PR to `development` branch.

**Comment by sunshinejr** [comment]
Cool, thanks @larryonoff! Can you just add an entry to Changelog? Here you can find some guidelines for that (this is also a breaking change).

**Comment by larryonoff** [comment]
@sunshinejr done. hope that according guidelines.

I'm thinking about making `ReactiveMoya` as something separate from main `Moya` project. So that we're able updating it independently.

File: Changelog.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Next
+- Updated minimum version of `ReactiveSwift` to 3.0.
```
Comment (sunshinejr) [inline Changelog.md]: We're almost there 😄 We need a *Breaking change* alias before the entry, and a column (probably `Changed` is the most fitting):
```
# Next
### Changed
- **Breaking Change** Updated minimum version of `ReactiveSwift` to 3.0.
```

Comment (larryonoff) [reply Changelog.md]: done :)

**Comment by lucas34** [comment]
Package.Resolved has not been updated. It's still pointing to 2.0.1.
Should run 'swift package update'

**Pull request merged**